src/SFA.DAS.EAS.Employer_Account.Database/StoredProcedures/CreateEmployerAgreement.sql

```diff
@@ -10,7 +11,7 @@ BEGIN
 	BEGIN
 		SELECT TOP 1 @templateId = Id
 		FROM [employer_account].[EmployerAgreementTemplate]
-		WHERE AgreementType = 'Levy'
+		WHERE AgreementType LIKE CASE @eoi WHEN 0 THEN 'Levy' ELSE 'NonLevy.EOI%' END
```

Comment:
Is this related to the 'magic string' discussion that was had around the api work?

Also, but this is wider than just this code review, are we happy that we could set `AgreementType` to 'NonLevy' here, but may well set the account Levy status to 'Levy' via the completely independent mechanism that sets that other flag?

Reply:
This scenario will be caught in the web app.

Reply:
We’re using the template (which has a version column on it) but we’re adding the version onto the end of the type and then doing like checks in stored procs? Is this some kind of elaborate wind up?

Reply:
This is in the design as signed off by the TA's.

Reply:
SFA.DAS.Authorization hasn't been migrated. CI are also using the same for their change.

Reply:
@MichaelYoung1981 Was this magic string thing part of the debate on the [api PR](https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/pull/1762)?

In which case is there much we can do about it here?

Reply:
I’ll speak to Roy on Monday but Dave’s requirement was simply that he didn’t want consumers of the api to require changes when a new version is introduced.
I’m aware you’re change is based on a previous stories implementation but that implementation shouldn’t have gone to master.

Reply:
@winsto yes, we can amend the original implementation if we feel it’s necessary to preserve the quality of the code base.

Reply:
@MichaelYoung1981 We'll hold off on this PR then until that discussion has been had.